src/autocompleteDump.ts

```diff
@@ -0,0 +1,124 @@
+import * as request from "request-promise-native"
+import * as vscode from "vscode"
+import { parseStringPromise } from "xml2js"
+
+const AUTOCOMPLETE_METADATA = "https://raw.githubusercontent.com/CloneTrooper1019/Roblox-Client-Tracker/roblox/AutocompleteMetadata.xml"
+
+export interface AutocompleteParameter {
+    type: string,
+    name: string,
+    optional: boolean
+}
+
+export interface AutocompleteReturn {
+    type: string,
+    name: string
+}
+
+export interface AutocompleteFunction {
+    name: string,
+    static: boolean,
+    description?: string,
+    parameters: AutocompleteParameter[]
+    returns: AutocompleteReturn[]
+}
+
+export interface AutocompleteProperty {
+    name: string,
+    static: boolean,
+    type: string,
+    description?: string
+}
+
+export interface AutocompleteGroup {
+    name: string,
+    functions: AutocompleteFunction[],
+    properties: AutocompleteProperty[]
+}
+
+export interface AutocompleteDump {
+    LuaLibrary: AutocompleteGroup[],
+    ItemStruct: AutocompleteGroup[]
+}
+
+const formatFunction = (func: any): AutocompleteFunction => {
+    const attributes = func.$
+    const parameters: AutocompleteParameter[] = 
+        (func.parameters && func.parameters[0].$$) ? func.parameters[0].$$.map((paramObj: any) => {
+            const type = paramObj["#name"]
+            const parameterAttributes = paramObj.$
+            return {
+                name: parameterAttributes.name,
+                optional: Boolean(parameterAttributes.optional) || false,
+                type,
+            }
+        }) : []
+
+    const returns = (func.returns && func.returns[0].$$) ? func.returns[0].$$.map((ret: any) => {
+        const type = ret["#name"]
+        const returnAttributes = ret.$
+        return {
+            name: returnAttributes ? returnAttributes.name : null,
+            type,
+        }
+    }) : []
+
+    return {
+        description: func.description ? func.description[0] : undefined,
+        name: attributes.name,
+        parameters,
+        returns,
+        static: Boolean(attributes.static) || false,
```

Comment:
What's with the call to `Boolean`?

Reply:
attributes.static is actually a string when parsed by xml2js, so I need to typecast it. I’m not actually sure if I need the `|| false` part though

Reply:
You can just do `attributes.static === "true"` and it'd be cleaner